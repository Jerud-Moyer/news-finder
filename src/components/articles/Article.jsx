import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ source, author, title, content }) => (
  <>
    <span>{source}</span>
    <span>{title}</span>
    <span>{author}</span>
    <div>{content}</div>
  </>
);

Article.propTypes = {
  source: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
};


export default Article;
