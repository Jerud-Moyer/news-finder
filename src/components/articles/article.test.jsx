import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());

  it('renders Article', () => {
    const { asFragment } = render(<Article
      source="Test Source"
      author="Test Author"
      title="Test Title"
      content="Test Content"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
