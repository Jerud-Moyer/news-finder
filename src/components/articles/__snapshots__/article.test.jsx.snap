// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <span>
    Test Source
  </span>
  <span>
    Test Title
  </span>
  <span>
    Test Author
  </span>
  <div>
    Test Content
  </div>
</DocumentFragment>
`;
